plugin/trino-hive/src/test/java/io/trino/plugin/hive/AbstractTestHive.java

```diff
@@ -3781,6 +3785,68 @@ public void testMaterializedViewMetadata()
         }
     }
 
+    @Test
+    public void testOrcPageSourceMetrics()
+            throws Exception
+    {
+        SchemaTableName tableName = temporaryTable("orc_page_source_metrics");
+        try {
+            assertPageSourceMetrics(tableName, ORC, new Metrics(ImmutableMap.of(ORC_CODEC_METRIC_PREFIX + "SNAPPY", new LongCount(209))));
+        }
+        finally {
+            dropTable(tableName);
+        }
+    }
+
+    @Test
+    public void testParquetPageSourceMetrics()
+            throws Exception
+    {
+        SchemaTableName tableName = temporaryTable("parquet_page_source_metrics");
+        try {
+            assertPageSourceMetrics(tableName, PARQUET, new Metrics(ImmutableMap.of(PARQUET_CODEC_METRIC_PREFIX + "SNAPPY", new LongCount(1169))));
+        }
+        finally {
+            dropTable(tableName);
+        }
+    }
+
+    private void assertPageSourceMetrics(SchemaTableName tableName, HiveStorageFormat storageFormat, Metrics expectedMetrics)
+            throws Exception
+    {
+        createEmptyTable(
+                tableName,
+                storageFormat,
+                ImmutableList.of(
+                        new Column("id", HIVE_LONG, Optional.empty()),
+                        new Column("name", HIVE_STRING, Optional.empty())),
+                ImmutableList.of());
+        MaterializedResult.Builder inputDataBuilder = MaterializedResult.resultBuilder(SESSION, BIGINT, VARCHAR);
+        IntStream.range(0, 100).forEach(i -> inputDataBuilder.row((long) i, String.valueOf(i)));
+        insertData(tableName, inputDataBuilder.build(), ImmutableMap.of("compression_codec", "SNAPPY"));
+
+        try (Transaction transaction = newTransaction()) {
+            ConnectorMetadata metadata = transaction.getMetadata();
+            ConnectorSession session = newSession();
+            metadata.beginQuery(session);
+
+            ConnectorTableHandle tableHandle = getTableHandle(metadata, tableName);
+
+            // read entire table
+            List<ColumnHandle> columnHandles = ImmutableList.<ColumnHandle>builder()
+                    .addAll(metadata.getColumnHandles(session, tableHandle).values())
+                    .build();
+
+            List<ConnectorSplit> splits = getAllSplits(getSplits(splitManager, transaction, session, tableHandle));
+            for (ConnectorSplit split : splits) {
+                try (ConnectorPageSource pageSource = pageSourceProvider.createPageSource(transaction.getTransactionHandle(), session, split, tableHandle, columnHandles, DynamicFilter.EMPTY)) {
+                    materializeSourceDataStream(session, pageSource, getTypes(columnHandles));
+                    assertThat(pageSource.getMetrics()).isEqualTo(expectedMetrics);
```

Comment:
Please check by running this with a high invocation count that relying on this number to be exactly the same each time is not flaky

Reply:
I did that locally. Its not flaky